consensus/beacon/consensus.go

```diff
@@ -74,16 +73,16 @@ func New(ethone consensus.Engine) *Beacon {
 	return &Beacon{ethone: ethone}
 }
 
-// TestingTTDBlock is a replacement mechanism for TTD-based pre-/post-merge
-// splitting. With chain history deletion, TD calculations become impossible.
-// This is fine for progressing the live chain, but to be able to generate test
-// chains, we do need a split point. This method supports setting an explicit
-// block number to use as the splitter *for testing*, instead of having to keep
-// the notion of TDs in the client just for testing.
-//
-// The block with supplied number is regarded as the last pre-merge block.
-func (beacon *Beacon) TestingTTDBlock(number uint64) {
-	beacon.ttdblock = &number
+// isPostMerge reports whether the given block number is assumed to be post-merge.
+// Here we check the MergeNetsplitBlock to allow configuring networks with a PoW or
+// PoA chain for unit testing purposes.
+func isPostMerge(config *params.ChainConfig, blockNum uint64, timestamp uint64) bool {
+	mergedAtGenesis := config.TerminalTotalDifficulty != nil && config.TerminalTotalDifficulty.Sign() == 0
+	return mergedAtGenesis ||
+		config.MergeNetsplitBlock != nil && blockNum >= config.MergeNetsplitBlock.Uint64() ||
+		config.ShanghaiTime != nil && timestamp >= *config.ShanghaiTime ||
+		// If OP-Stack then bedrock activation number determines when TTD (eth Merge) has been reached.
+		config.Optimism != nil && config.IsBedrock(new(big.Int).SetUint64(blockNum))
```

Comment:
```suggestion
		config.IsOptimismBedrock(new(big.Int).SetUint64(blockNum))
```
There's a shorthand for that